### ISSUE TITLE
Crash when exiting playback, v0.14.0-beta1, exoplayer

### ISSUE THREAD

**Issue by amasover** [jellyfin/jellyfin-androidtv#1855]
### Describe the bug

I can't seem to reproduce this reliably, but the crash seems to happen most often when stopping playback of a video.

1. Start playback
2. Change audio tracks maybe? Or skip around in the video?
3. Exit video playback
4. Crash?

The video from the attached crash log had the following characteristics:

General:
Format: Matroska
Duration: 1 h 39 min
Global Bit Rate: 82.1mb/s
Overall Size: 56.9 GiB

Video Tracks:
Track 1:
Format: HEVC (10 bits)
Resolution: 3840 x 2160
Aspect Ratio: 16:9
Frame Rate: 23.976 (24000/1001) FPS
Bit Rate: 78.4mb/s
HDR Format: SMPTE ST 2094 App 4, Version 1, HDR10+ Profile B compatible
Color Primaries: BT.2020
Transfer Characteristics: PQ

Audio Tracks:
Track 1:
English | MLP FBA | 5.1ch | 1227kb/s | Discrete Surround Upmix / Dolby TrueHD Audio / 5.1 / 48 kHz / 1227 kbps / 16-bit
Track 2:
English | AC-3 | 5.1ch | 640kb/s | Compatibility Track / Dolby Digital Audio / 5.1-EX / 48 kHz / 640 kbps / DN -27dB
Track 3:
English | FLAC | 2.0ch | 1184kb/s | Dolby SVA 35mm / FLAC Audio / Lt/Rt / 48 kHz / 1184 kbps / 24-bit
Track 4:
English | AC-3 | 2.0ch | 640kb/s | Commentary

I have also noticed that when I seek, especially right after just beginning to play a video, sometimes the app will hang for around 30 seconds, and sometimes I will see a toast notification that the player encountered an error and had to be reloaded. Also, seeking seems to often cause the player to reload the audio/video (I can tell it's being reloaded because with my setup initializing the audio can cause a *pop* from my speakers). I'm not sure if any of that is related to this issue though.

### Logs

---
client: Jellyfin for Android TV
client_version: 0.14.0-beta.1
client_repository: https://github.com/jellyfin/jellyfin-androidtv
type: crash_report
format: markdown
---
### Logs

***Stack Trace***: 
```log
java.lang.IllegalStateException: can't get VLCObject instance
	at org.videolan.libvlc.MediaPlayer.nativeSetAudioOutputDevice(Native Method)
	at org.videolan.libvlc.MediaPlayer.setAudioOutputDeviceInternal(MediaPlayer.java:969)
	at org.videolan.libvlc.MediaPlayer.updateAudioOutputDevice(MediaPlayer.java:459)
	at org.videolan.libvlc.MediaPlayer.access$600(MediaPlayer.java:58)
	at org.videolan.libvlc.MediaPlayer$3.onAudioDevicesChanged(MediaPlayer.java:535)
	at org.videolan.libvlc.MediaPlayer$3.onAudioDevicesRemoved(MediaPlayer.java:559)
	at android.media.AudioManager$NativeEventHandlerDelegate$1.handleMessage(AudioManager.java:6270)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7668)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

```  
***Logcat***: 
```log
--------- beginning of main
07-17 16:53:06.436 D/jellyfin-apiclient(23095): Adding request to queue: http://192.168.1.213:8096/Sessions/Playing/Progress
07-17 16:53:06.523 I/jellyfin-apiclient(23095): Response received from: http://192.168.1.213:8096/Sessions/Playing/Progress
07-17 16:53:06.554 D/ACRA    (23095): onActivityPaused class org.jellyfin.androidtv.ui.playback.PlaybackOverlayActivity
07-17 16:53:06.585 D/ACRA    (23095): onActivityStarted class org.jellyfin.androidtv.ui.itemdetail.FullDetailsActivity
07-17 16:53:06.587 D/ACRA    (23095): onActivityResumed class org.jellyfin.androidtv.ui.itemdetail.FullDetailsActivity
07-17 16:53:06.588 D/BackgroundServiceFragment(23095): Restoring active backgrounds
07-17 16:53:06.588 D/BackgroundServiceFragment(23095): Restoring background drawable
07-17 16:53:06.648 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_ASYNC"
07-17 16:53:06.648 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_LOGFPS"
07-17 16:53:06.648 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_PROCESS_NAME"
07-17 16:53:06.648 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_FPSDYNAMIC"
07-17 16:53:07.142 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_ASYNC"
07-17 16:53:07.142 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_LOGFPS"
07-17 16:53:07.142 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_PROCESS_NAME"
07-17 16:53:07.142 E/libc    (23095): Access denied finding property "ro.vendor.sys.NV_FPSDYNAMIC"
07-17 16:53:07.566 D/SurfaceUtils(23095): connecting to surface 0x26cb35e520, reason connectToSurface
07-17 16:53:07.566 I/MediaCodec(23095): [OMX.Nvidia.h265.decode] setting surface generation to 23649291
07-17 16:53:07.566 D/SurfaceUtils(23095): disconnecting from surface 0x26cb35e520, reason connectToSurface(reconnect)
07-17 16:53:07.566 D/SurfaceUtils(23095): connecting to surface 0x26cb35e520, reason connectToSurface(reconnect)
07-17 16:53:07.834 D/SurfaceUtils(23095): set up nativeWindow 0x26cb35e520 for 3840x2160, color 0x18b, rotation 0, usage 0x2900
07-17 16:53:07.834 D/SurfaceUtils(23095): disconnecting from surface 0x26cb3518c0, reason disconnectFromSurface
07-17 16:53:07.835 E/Surface (23095): freeAllBuffers: 12 buffers were freed while being dequeued!
07-17 16:53:07.841 E/Surface (23095): freeAllBuffers: 1 buffers were freed while being dequeued!
07-17 16:53:07.852 E/AudioTrack-JNI(23095): Error -32 during AudioTrack native read
07-17 16:53:07.852 E/DecoderAudioRenderer(23095): Audio sink error
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):   com.google.android.exoplayer2.audio.AudioSink$WriteException: AudioTrack write failed: -6
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.audio.DefaultAudioSink.writeBuffer(DefaultAudioSink.java:1166)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.audio.DefaultAudioSink.processBuffers(DefaultAudioSink.java:1090)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.audio.DefaultAudioSink.handleBuffer(DefaultAudioSink.java:1022)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.processOutputBuffer(MediaCodecAudioRenderer.java:703)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.bypassRender(MediaCodecRenderer.java:2180)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:787)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:989)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:490)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at android.os.Handler.dispatchMessage(Handler.java:102)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at android.os.Looper.loop(Looper.java:223)
07-17 16:53:07.852 E/DecoderAudioRenderer(23095):       at android.os.HandlerThread.run(HandlerThread.java:67)
07-17 16:53:07.856 D/ACRA    (23095): onActivityStopped class org.jellyfin.androidtv.ui.playback.PlaybackOverlayActivity
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095): Recoverable renderer error
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):   com.google.android.exoplayer2.ExoPlaybackException: MediaCodecAudioRenderer error, index=1, format=Format(2, Discrete Surround Upmix / Dolby TrueHD Audio / 5.1 / 48 kHz / 1227 kbps / 16-bit, null, audio/true-hd, null, -1, en, [-1, -1, -1.0], [6, 48000]), format_supported=YES
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:566)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at android.os.Handler.dispatchMessage(Handler.java:102)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at android.os.Looper.loop(Looper.java:223)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at android.os.HandlerThread.run(HandlerThread.java:67)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):   Caused by: com.google.android.exoplayer2.audio.AudioSink$WriteException: AudioTrack write failed: -6
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.audio.DefaultAudioSink.writeBuffer(DefaultAudioSink.java:1166)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.audio.DefaultAudioSink.processBuffers(DefaultAudioSink.java:1090)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.audio.DefaultAudioSink.handleBuffer(DefaultAudioSink.java:1022)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.audio.MediaCodecAudioRenderer.processOutputBuffer(MediaCodecAudioRenderer.java:703)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.bypassRender(MediaCodecRenderer.java:2180)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:787)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:989)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:490)
07-17 16:53:07.856 W/ExoPlayerImplInternal(23095):       ... 3 more
07-17 16:53:07.858 D/CustomPlaybackOverlayFragment(23095): Stopping!
07-17 16:53:07.858 D/CustomPlaybackOverlayFragment(23095): this fragment belongs to the current session, ending it
07-17 16:53:07.858 D/PlaybackController(23095): stop called at 2525363
07-17 16:53:07.884 D/SurfaceUtils(23095): connecting to surface 0x26cb346e70, reason connectToSurface
07-17 16:53:07.884 I/MediaCodec(23095): [OMX.Nvidia.h265.decode] setting surface generation to 23649292
07-17 16:53:07.884 D/SurfaceUtils(23095): disconnecting from surface 0x26cb346e70, reason connectToSurface(reconnect)
07-17 16:53:07.884 D/SurfaceUtils(23095): connecting to surface 0x26cb346e70, reason connectToSurface(reconnect)
07-17 16:53:07.885 D/SurfaceUtils(23095): set up nativeWindow 0x26cb346e70 for 3840x2160, color 0x18b, rotation 0, usage 0x2900
07-17 16:53:07.886 D/SurfaceUtils(23095): disconnecting from surface 0x26cb35e520, reason disconnectFromSurface
07-17 16:53:07.886 E/Surface (23095): freeAllBuffers: 12 buffers were freed while being dequeued!
07-17 16:53:07.908 D/SurfaceUtils(23095): disconnecting from surface 0x26cb346e70, reason disconnectFromSurface
07-17 16:53:07.978 W/AudioSystem(23095): ioConfigChanged() modifying unknown output! 790
07-17 16:53:07.980 W/AudioSystem(23095): ioConfigChanged() modifying unknown output! 5101
07-17 16:53:08.009 I/jellyfin-apiclient(23095): ReportPlaybackStopped: Item 1757febc5785a9813f721f2062937bb5, Ticks: 25253630000
07-17 16:53:08.010 D/jellyfin-apiclient(23095): Adding request to queue: http://192.168.1.213:8096/Sessions/Playing/Stopped
07-17 16:53:08.012 I/ExoPlayerImpl(23095): Release 2e49572 [ExoPlayerLib/2.18.0] [mdarcy, SHIELD Android TV, NVIDIA, 30] [goog.exo.core, goog.exo.ui, goog.exo.exoplayer, goog.exo.decoder, goog.exo.datasource, goog.exo.extractor, goog.exo.ffmpeg]
07-17 16:53:08.016 D/ACRA    (23095): onActivityDestroyed class org.jellyfin.androidtv.ui.playback.PlaybackOverlayActivity
07-17 16:53:08.041 D/FullDetailsActivity(23095): current time 1658098388041 last playback event time 1658098388010 last refresh time 1658098119319
07-17 16:53:08.041 D/FullDetailsActivity(23095): Updating info after playback
07-17 16:53:08.042 D/jellyfin-apiclient(23095): Adding request to queue: http://192.168.1.213:8096/Users/c3a0e26d-d1ba-4933-a065-cc660bc217ec/Items/1757febc5785a9813f721f2062937bb5?format=json
07-17 16:53:08.042 D/AndroidRuntime(23095): Shutting down VM
--------- beginning of crash
07-17 16:53:08.068 E/AndroidRuntime(23095): FATAL EXCEPTION: main
07-17 16:53:08.068 E/AndroidRuntime(23095): Process: org.jellyfin.androidtv, PID: 23095
07-17 16:53:08.068 E/AndroidRuntime(23095): java.lang.IllegalStateException: can't get VLCObject instance
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at org.videolan.libvlc.MediaPlayer.nativeSetAudioOutputDevice(Native Method)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at org.videolan.libvlc.MediaPlayer.setAudioOutputDeviceInternal(MediaPlayer.java:969)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at org.videolan.libvlc.MediaPlayer.updateAudioOutputDevice(MediaPlayer.java:459)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at org.videolan.libvlc.MediaPlayer.access$600(MediaPlayer.java:58)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at org.videolan.libvlc.MediaPlayer$3.onAudioDevicesChanged(MediaPlayer.java:535)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at org.videolan.libvlc.MediaPlayer$3.onAudioDevicesRemoved(MediaPlayer.java:559)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at android.media.AudioManager$NativeEventHandlerDelegate$1.handleMessage(AudioManager.java:6270)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at android.os.Looper.loop(Looper.java:223)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at android.app.ActivityThread.main(ActivityThread.java:7668)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at java.lang.reflect.Method.invoke(Native Method)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
07-17 16:53:08.068 E/AndroidRuntime(23095): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
07-17 16:53:08.068 E/ACRA    (23095): ACRA caught a IllegalStateException for org.jellyfin.androidtv
07-17 16:53:08.068 E/ACRA    (23095): java.lang.IllegalStateException: can't get VLCObject instance
07-17 16:53:08.068 E/ACRA    (23095): 	at org.videolan.libvlc.MediaPlayer.nativeSetAudioOutputDevice(Native Method)
07-17 16:53:08.068 E/ACRA    (23095): 	at org.videolan.libvlc.MediaPlayer.setAudioOutputDeviceInternal(MediaPlayer.java:969)
07-17 16:53:08.068 E/ACRA    (23095): 	at org.videolan.libvlc.MediaPlayer.updateAudioOutputDevice(MediaPlayer.java:459)
07-17 16:53:08.068 E/ACRA    (23095): 	at org.videolan.libvlc.MediaPlayer.access$600(MediaPlayer.java:58)
07-17 16:53:08.068 E/ACRA    (23095): 	at org.videolan.libvlc.MediaPlayer$3.onAudioDevicesChanged(MediaPlayer.java:535)
07-17 16:53:08.068 E/ACRA    (23095): 	at org.videolan.libvlc.MediaPlayer$3.onAudioDevicesRemoved(MediaPlayer.java:559)
07-17 16:53:08.068 E/ACRA    (23095): 	at android.media.AudioManager$NativeEventHandlerDelegate$1.handleMessage(AudioManager.java:6270)
07-17 16:53:08.068 E/ACRA    (23095): 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-17 16:53:08.068 E/ACRA    (23095): 	at android.os.Looper.loop(Looper.java:223)
07-17 16:53:08.068 E/ACRA    (23095): 	at android.app.ActivityThread.main(ActivityThread.java:7668)
07-17 16:53:08.068 E/ACRA    (23095): 	at java.lang.reflect.Method.invoke(Native Method)
07-17 16:53:08.068 E/ACRA    (23095): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
07-17 16:53:08.068 E/ACRA    (23095): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
07-17 16:53:08.068 D/ACRA    (23095): Building report
07-17 16:53:08.072 D/ACRA    (23095): Starting collectors with priority FIRST
07-17 16:53:08.073 D/ACRA    (23095): Calling collector org.acra.collector.DropBoxCollector
07-17 16:53:08.073 D/ACRA    (23095): Calling collector org.acra.collector.LogCatCollector
07-17 16:53:08.073 D/ACRA    (23095): Collector org.acra.collector.DropBoxCollector completed
07-17 16:53:08.075 D/ACRA    (23095): Calling collector org.acra.collector.StacktraceCollector
07-17 16:53:08.075 D/ACRA    (23095): Collector org.acra.collector.StacktraceCollector completed
07-17 16:53:08.077 D/ACRA    (23095): Retrieving logcat output (buffer:default)...

```  

### App information

***App version***: `0.14.0-beta.1` (`140001`)  
***Package name***: `org.jellyfin.androidtv`  
***Build***: 
```json
{"BOARD":"darcy","BOOTLOADER":"32.00.2019.50-t210-69ebfcbe","BRAND":"NVIDIA","CPU_ABI":"arm64-v8a","CPU_ABI2":"","DEVICE":"mdarcy","DISPLAY":"RQ1A.210105.003.7825230_3062.3935","FINGERPRINT":"NVIDIA\/mdarcy\/mdarcy:11\/RQ1A.210105.003\/7825230_3062.3935:user\/release-keys","HARDWARE":"darcy","HOST":"mobile-u64-5417-d8000","ID":"RQ1A.210105.003","IS_DEBUGGABLE":false,"IS_EMULATOR":false,"MANUFACTURER":"NVIDIA","MODEL":"SHIELD Android TV","PERMISSIONS_REVIEW_REQUIRED":true,"PRODUCT":"mdarcy","RADIO":"unknown","SUPPORTED_32_BIT_ABIS":["armeabi-v7a","armeabi"],"SUPPORTED_64_BIT_ABIS":["arm64-v8a"],"SUPPORTED_ABIS":["arm64-v8a","armeabi-v7a","armeabi"],"TAGS":"release-keys","TIME":1655329637000,"TYPE":"user","UNKNOWN":"unknown","USER":"NVIDIA","VERSION":{"ACTIVE_CODENAMES":[],"BASE_OS":"","CODENAME":"REL","INCREMENTAL":"7825230_3062.3935","PREVIEW_SDK_FINGERPRINT":"REL","PREVIEW_SDK_INT":0,"RELEASE":"11","RELEASE_OR_CODENAME":"11","SDK":"30","SDK_INT":30,"SECURITY_PATCH":"2022-04-05"}}
```  
***Build config***: 
```json
{"APPLICATION_ID":"org.jellyfin.androidtv","BUILD_TYPE":"release","DEBUG":false,"DEVELOPMENT":false,"VERSION_CODE":140001,"VERSION_NAME":"0.14.0-beta.1"}
```  

### Device information

***Android version***: `11`  
***Device brand***: `NVIDIA`  
***Device product***: `mdarcy`  
***Device model***: `SHIELD Android TV`  

### Crash information

***Start time***: `2022-07-17T16:42:58.878-06:00`  
***Crash time***: `2022-07-17T16:53:08.104-06:00`  

### Application version

0.14.0-beta1

### Where did you install the app from?

Google Play

### Device information

Nvidia Shield Pro (2019)

### Android version

Android 11

### Jellyfin server version

10.8.1

**Comment by nielsvanvelzen** [comment]
Is this issue new in 0.14 or did you also experience it in 0.13?

**Comment by luckkmaxx** [comment]
I think the title has a wrong and unexisting (yet) version.

**Comment by nielsvanvelzen** [comment]
0.14-beta.1 is definitely a real version.

**Comment by amasover** [comment]
Apologies for the typo in the title.

I rolled back to 0.13, and I can reproduce this issue. I noticed that I seem to only be able to reproduce it when refresh rate switching is enabled, so this issue may be a dupe of #1787.

For me when the issue happens, it doesn't kick me back to the login screen, but it does cause the app to restart in 0.13. In 0.14.0-beta1 the app restarts and opens a crash log reporting window.

I think the seeking issue I mentioned that leads to the "player encountered an error" toast might be unrelated, as that seems to happen regardless of whether refresh rate switching is enabled or not. So I may open a separate issue for that if it's unrelated.
